src/Arcus.Template.WebApi/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM mcr.microsoft.com/dotnet/core/aspnet:2.2-stretch-slim AS base
+WORKDIR /app
+EXPOSE 80
+
+FROM mcr.microsoft.com/dotnet/core/sdk:2.2-stretch AS build
+WORKDIR /src
+COPY ["Arcus.Template.WebApi/Arcus.Template.WebApi.csproj", "Arcus.Template.WebApi/"]
+RUN dotnet restore "Arcus.Template.WebApi/Arcus.Template.WebApi.csproj"
+COPY . .
+WORKDIR "/src/Arcus.Template.WebApi"
+RUN dotnet build "Arcus.Template.WebApi.csproj" -c Release
+RUN dotnet new -i "bin/Release/Arcus.Template.WebApi.*.nupkg"
```

Comment:
It feels like the build part is two-fold:
- Build our template to generate a new NuGet package
- Build a new instance of our template, based on the NuGet package

Once that is done, we publish the new project.

Wouldn't it be better to have to stages:
- "Template Package Creation"
- "Project Generation & Publish", based on "Template Package Creation"

Reply:
By stages, you mean YAML stages, therefore two docker files? Not entirely sure what you mean?

Reply:
No a Docker stage - Everytime a Dockerfile has a `FROM` it is the start of a new stage - https://docs.docker.com/develop/develop-images/multistage-build/

But we can talk about this tomorrow if you want.

Reply:
Resolving as it's no longer applicable